challenges-31-plus.yml

```diff
@@ -0,0 +1,85 @@
+frame_redirect:
+  question: |
+    Our website has a very special Analytics tool made by us that does *not* use
+    any cookies! Our tool has a very convenient Symfony Package that we can use
+    while processing our requests.
+    
+    In our last weekly meeting, our boss told us that we need to log the name
+    of the frame making every request using this tool.
+    
+    How can we get this information form a controller?
```

Comment:
Yea, I see what you mean. What about:

> On our cart page, we have a `<turbo-frame id="cart-summary">` that renders a short list of the items in our
> cart. This works great! But now, in our base layout, we want to render a link on our sidebar that, on click, loads this "cart summary" *into* the sidebar:

```rst
<turbo-frame id="sidebar">
    <!-- ... -->

    <a href="{{ path('app_cart') }}">Show Cart Summary</a>
</turbo>
```

> Unfortunately, when we try this, we get an error because the `turbo-frame` looks for a `sidebar` frame on
> the cart page. How could me modify the cart page to get this to work?

Then, the solution is to read the `turbo-frame` header in cart controller (with a default of `cart-summary` if there is no header) and render that dynamic frame value in its template. We'll need to think of a couple of other wrong answers :)